### DISCUSSION TITLE
Failed to open datastore error

### DISCUSSION THREAD

**Discussion by Greg-I-Anderson** [praetorian-inc/noseyparker#92] [Q&A]
Anyone know why I am getting the following error when running scans? - 
Error: Failed to open datastore at np.my_repo

Caused by:
    0: Failed to open database at np.my_repo/datastore.db
    1: database is locked
    2: Error code 5: The database file is locked

I get the same error when using the pre-built binary or Docker.

The specific Docker command I was using was - 
docker run -v "$PWD":/scan ghcr.io/praetorian-inc/noseyparker:latest scan --datastore np.my_repo my_repo.git

**Comment by bradlarsen** [comment]
Interesting.

@Greg-I-Anderson, do you have some other process accessing the datastore, like a `sqlite3` interpreter or some other app? That could cause what you're seeing.

What happens if you try with a new datastore that definitely does not exist, e.g.,
```
docker run -v "$PWD":/scan ghcr.io/praetorian-inc/noseyparker:latest scan --datastore some_nonexistent_datastore.np my_repo.git
```
Same error?

**Reply by Greg-I-Anderson** [reply]
Thanks very much for your response.  I am not aware of any other process accessing the datastore.  I have tried the command with a new datastore that does not exist and I get the same error.   I asked a colleague to try it too - and he gets the same error.

**Reply by bradlarsen** [reply]
What OS are you running on?

If you try the following (note the different location of the datastore), do you get the same error?
```
docker run -v "$PWD":/scan ghcr.io/praetorian-inc/noseyparker:latest scan --datastore /root/some_nonexistent_datastore.np my_repo.git
```

**Reply by bradlarsen** [reply]
Also, what does `docker run ghcr.io/praetorian-inc/noseyparker:latest --version` print out?

**Reply by munntjlx** [reply]
also try "lsof (assuming you have installed it) | grep -i your file

**Reply by Greg-I-Anderson** [reply]
Thanks both for your responses.  
I am using a Microsoft Azure compute instance running on Linux.
Version -
noseyparker 0.15.0

Build Configuration:

    Build Timestamp:    2023-10-13T03:58:20.722595058Z

    Commit Timestamp:   2023-10-12T22:58:14.000000000-04:00
    Commit Branch:      HEAD
    Commit SHA:         133aa4ef01169d3d21f280f3c3de214e9caa299b

    Cargo Features:     disable_trace,log,release
    Debug:              false
    Optimization:       3
    Target Triple:      x86_64-unknown-linux-gnu

Build System:

    OS:                 Debian GNU/Linux
    OS Version:         Linux 11 Debian GNU/Linux

    CPU Vendor:         GenuineIntel
    CPU Brand:          Intel(R) Xeon(R) CPU E5-2673 v3 @ 2.40GHz
    CPU Cores:          2

    rustc Version:      1.71.1
    rustc Channel:      stable
    rustc Host Triple:  x86_64-unknown-linux-gnu
    rustc Commit Date:  2023-08-03
    rustc Commit SHA:   eb26296b556cef10fb713a38f3d16b9886080f26
    rustc LLVM Version: 16.0

**Reply by bradlarsen** [reply]
@Greg-I-Anderson If you run the following (note the different path to the datastore here), do you still see the same "database is locked" error?
```
docker run -v "$PWD":/scan ghcr.io/praetorian-inc/noseyparker:latest scan --datastore /root/some_nonexistent_datastore.np my_repo.git
```

**Reply by Greg-I-Anderson** [reply]
docker run -v "$PWD":/scan ghcr.io/praetorian-inc/noseyparker:latest scan --datastore /root/screening.np screening.git
2023-10-30T09:13:33.580831Z ERROR noseyparker::input_enumerator: Failed to get entry: screening.git: No such file or directory (os error 2); skipping
Scanned 0 B from 0 blobs in 0 seconds (0 B/s); 0/0 new matches

**Reply by bradlarsen** [reply]
@Greg-I-Anderson okay, that Docker output does show a different error. In that case, Nosey Parker actually ran without running into that datastore error, but the repository you were trying to scan `screening.git` didn't exist, and nothing was actually scanned.

The "database is locked" error you have been seeing is still mysterious to me.
